packages/menu/src/Menu.ts

```diff
@@ -201,16 +345,43 @@ export class Menu extends SpectrumElement {
             item = this.menuItems[index] as MenuItem;
         }
         index = Math.max(index, 0);
+        const selectedItemsMap = new Map<MenuItem, boolean>();
+        const selected: string[] = [];
+        const selectedItems: MenuItem[] = [];
         this.menuItems.forEach((item, i) => {
+            if (item.selected) {
+                selectedItemsMap.set(item, true);
+                selected.push(item.value);
+                selectedItems.push(item);
+            }
             if (i !== index) {
                 item.focused = false;
             }
         });
+        this.selectedItemsMap = selectedItemsMap;
+        this.selected = selected;
+        this.selectedItems = selectedItems;
         this.focusedItemIndex = index;
         this.focusInItemIndex = index;
     }
 
-    private prepItems = (): void => {
+    private async updateItemRoles() {
+        const childMenuItems = [
+            ...this.querySelectorAll(`sp-menu-item`),
+        ] as MenuItem[];
+
+        const promises: Promise<any>[] = [];
+
+        for (const item of childMenuItems) {
+            item.updateRole();
+            promises.push(item.updateComplete);
+        }
+        await Promise.all(promises);
```

Comment:
I'm guessing this should be removed once we properly resolve this.

Reply:
Yes, especially because `role` isn't a watched property, so without a `this.requestUpdate()` or something there won't technically be work happening in the child items here.